Runtimes/Supplemental/cmake/modules/FindSwiftCore.cmake

```diff
@@ -52,112 +52,102 @@ endif()
-    INTERFACE_INCLUDE_DIRECTORIES "${SwiftCore_INCLUDE_DIR}")
-  find_package_handle_standard_args(SwiftCore DEFAULT_MSG
-    SwiftCore_IMPLIB SwiftCore_INCLUDE_DIR)
+  list(APPEND SwiftCore_NAMES libswiftCore.tbd)
 elseif(LINUX)
-  if (NOT BUILD_SHARED_LIBS)
-    find_path(SwiftCore_INCLUDE_DIR
-      "Swift.swiftmodule"
-      HINTS
-        "${Swift_SDKROOT}/usr/lib/swift_static/linux-static")
-    find_library(SwiftCore_LIBRARY
-      NAMES "libswiftCore.a"
-      HINTS "${Swift_SDKROOT}/usr/lib/swift_static/linux-static")
-    add_library(swiftCore STATIC IMPORTED GLOBAL)
+  if (SwiftCore_USE_STATIC_LIBS)
+    list(APPEND SwiftCore_INCLUDE_DIR_HINTS
+      "${Swift_SDKROOT}/usr/lib/swift_static/linux-static")
+    list(APPEND SwiftCore_LIBRARY_HINTS
+      "${Swift_SDKROOT}/usr/lib/swift_static/linux-static")
+    list(APPEND SwiftCore_NAMES libswiftCore.a)
   else()
-    find_path(SwiftCore_INCLUDE_DIR
-      "Swift.swiftmodule"
-      HINTS
-        "${Swift_SDKROOT}/usr/lib/swift/linux")
-    find_library(SwiftCore_LIBRARY
-      NAMES "libswiftCore.so"
-      HINTS "${Swift_SDKROOT}/usr/lib/swift/linux")
-    add_library(swiftCore SHARED IMPORTED GLOBAL)
+    list(APPEND SwiftCore_INCLUDE_DIR_HINTS
+      "${Swift_SDKROOT}/usr/lib/swift/linux")
+    list(APPEND SwiftCore_LIBRARY_HINTS
+      "${Swift_SDKROOT}/usr/lib/swift/linux")
+    list(APPEND SwiftCore_NAMES libswiftCore.so)
   endif()
-  set_target_properties(swiftCore PROPERTIES
-    IMPORTED_LOCATION "${SwiftCore_LIBRARY}"
-    INTERFACE_INCLUDE_DIRECTORIES "${SwiftCore_INCLUDE_DIR}")
-  find_package_handle_standard_args(SwiftCore DEFAULT_MSG
-    SwiftCore_LIBRARY SwiftCore_INCLUDE_DIR)
 elseif(WIN32)
-  find_path(SwiftCore_INCLUDE_DIR
-    "Swift.swiftmodule"
-    HINTS
-      "${Swift_SDKROOT}/usr/lib/swift/windows"
-      "$ENV{SDKROOT}/usr/lib/swift/windows")
-  find_library(SwiftCore_LIBRARY
-    NAMES "libswiftCore.lib"
-    HINTS
-      "${Swift_SDKROOT}/usr/lib/swift/${SwiftCore_PLATFORM_SUBDIR}/${SwiftCore_ARCH_SUBDIR}"
-      "${Swift_SDKROOT}/usr/lib/swift"
-      "$ENV{SDKROOT}/usr/lib/swift/${SwiftCore_PLATFORM_SUBDIR}/${SwiftCore_ARCH_SUBDIR}"
-      "$ENV{SDKROOT}/usr/lib/swift")
-
-  add_library(swiftCore SHARED IMPORTED GLOBAL)
-  set_target_properties(swiftCore PROPERTIES
-    IMPORTED_IMPLIB "${SwiftCore_LIBRARY}"
-    INTERFACE_INCLUDE_DIRECTORIES "${SwiftCore_INCLUDE_DIR}")
-  find_package_handle_standard_args(SwiftCore DEFAULT_MSG
-    SwiftCore_LIBRARY SwiftCore_INCLUDE_DIR)
+  list(APPEND SwiftCore_INCLUDE_DIR_HINTS
+    "${Swift_SDKROOT}/usr/lib/swift/windows"
+    "$ENV{SDKROOT}/usr/lib/swift/windows")
+  list(APPEND SwiftCore_LIBRARY_HINTS
+    "${Swift_SDKROOT}/usr/lib/swift/${SwiftCore_PLATFORM_SUBDIR}/${SwiftCore_ARCH_SUBDIR}"
+    "${Swift_SDKROOT}/usr/lib/swift"
+    "$ENV{SDKROOT}/usr/lib/swift/${SwiftCore_PLATFORM_SUBDIR}/${SwiftCore_ARCH_SUBDIR}"
+    "$ENV{SDKROOT}/usr/lib/swift")
+  list(APPEND SwiftCore_NAMES libswiftCore.lib)
```

Comment:
This is the static library name, `swiftCore.lib` is the import library. Please add the static case as well.

Reply:
Amended in [89b7929](https://github.com/swiftlang/swift/pull/82035/commits/89b7929bd51619543617184f6e619b32f0baa8ff) -- please note I did not touch the search paths, as I don't remember seeing any `swift_static` in my Swift installation for Windows.